core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/AvroUpsertMonitor.scala

```diff
@@ -378,9 +379,9 @@ class AvroUpsertMonitorActor(
         case Success(entityList) => entityList.size
       }.sum
 
-      val failureReports:List[RawlsErrorReport] = upsertResults collect {
-        case Failure(regrets:RawlsExceptionWithErrorReport) => regrets.errorReport.causes
-      } flatten
+      val failureReports: List[RawlsErrorReport] = upsertResults collect {
+        case Failure(regrets:RawlsExceptionWithErrorReport) => regrets.errorReport
+      }
```

Comment:
```suggestion
      // if the failure has underlying causes, use those; else use the parent failure.
      // unresolved entity references only have useful error messages in the underlying causes, not the parent failure
      // therefore to deliver the best messages to the user we need to handle both cases
      val failureReports: List[RawlsErrorReport] = upsertResults collect {
        case Failure(regrets:RawlsExceptionWithErrorReport) if regrets.errorReport.causes.nonEmpty => regrets.errorReport.causes
        case Failure(regrets:RawlsExceptionWithErrorReport) => Seq(regrets.errorReport)
      } flatten
```

your PR would cause a functional change, most importantly around unresolved entity references. The previous code would return a message like "entityType entityName not found", but your PR would make that less helpful, returning "Could not resolve some entity references". I think my suggestion resolves this for all cases.

NB: I suppose it would be even better if we could return the conjunction "Could not resolve some entity references: entityType entityName not found"!